src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/DeploymentOptions.cs

```diff
@@ -241,6 +244,52 @@ public DeploymentOptions()
         }
 
         public DeploymentOptions(DacDeployOptions options)
+        {
+            SetOptions(options);
+        }
+
+        /// <summary>
+        /// create deployment options from the options in a publish profile.xml
+        /// </summary>
+        /// <param name="options">options created from the profile</param>
+        /// <param name="profilePath"></param>
+        public DeploymentOptions(DacDeployOptions options, string profilePath)
+        {
+            // check if defaults need to be set if they aren't specified in the profile
+            string contents = File.ReadAllText(profilePath);
+            if (!contents.Contains("<AllowDropBlockingAssemblies>"))
+            {
+                options.AllowDropBlockingAssemblies = true;
+            }
+            if (!contents.Contains("<AllowIncompatiblePlatform>"))
+            {
+                options.AllowIncompatiblePlatform = true;
+            }
+            if (!contents.Contains("<DropObjectsNotInSource>"))
+            {
+                options.DropObjectsNotInSource = true;
```

Comment:
Double-checking that this (and the rest of the `true` crew) are actually supposed to be true here, 'cause they feel backward...

> If the profile _doesn't_ say DropObjectsNotInSource, then set DropObjectsNotInSource to true... which then drops them.

Am I reading that right?

Reply:
Yup the default for this setting is false, but when deploying we want to dropping the objects not in source to be the default. These are the same values that DacFx and SSMS deploy wizard set as the defaults for these settings.